File: dojo_11/java/src/main/java/steps/SeleccionDeEnvio.java
package steps;

public class SeleccionDeEnvio implements CheckoutStep{

	private final CheckoutStep nextStep;

	public SeleccionDeEnvio() {
		this.nextStep = new SeleccionDeMedioDePago();
	}

	public SeleccionDeEnvio(CheckoutStep nextStep) {
		this.nextStep = nextStep;
	}

	public CheckoutStep envioADomicilio() {
        return this.nextStep;
    }

    public CheckoutStep retiroEnCorreo() {
        return new MapaDeSucursales();
    }

	public CheckoutStep seleccionarEnvio(EnvioADomicilio envio){
		return new SeleccionDeMedioDePago(envio);
	}

	public Inconsistencia seleccionarEnvio(Express envio) {
		return new Inconsistencia();
	}
}

File: dojo_11/javascript/src/pasos/SeleccionDeEnvio.js
import PasoDeCheckout from './PasoDeCheckout';

import SeleccionDeMedioDePago from '../pasos/SeleccionDeMedioDePago';
import MapaDeSucursales from '../pasos/MapaDeSucursales';
import Review from './Review';

class SeleccionDeEnvio extends PasoDeCheckout {
    constructor(editMode) {
        super(editMode);
    }

    envioADomicilio() {
        if (!this.editMode) {
            return new SeleccionDeMedioDePago();
        } else {
            return new Review();
        }
    }

    retiroEnCorreo() {
        return new MapaDeSucursales();
    }
}

export default SeleccionDeEnvio;